packages/block-editor/src/components/rich-text/use-paste-handler.js

```diff
@@ -224,10 +224,14 @@ export function usePasteHandler( props ) {
 				mode = 'BLOCKS';
 			}
 
+			const trimmedPlainText = plainText.trim();
+
 			if (
 				__unstableEmbedURLOnPaste &&
 				isEmpty( value ) &&
-				isURL( plainText.trim() )
+				isURL( trimmedPlainText ) &&
+				// For the link pasting feature, allow only http(s) protocols.
+				/^https?:/.test( trimmedPlainText )
```

Comment:
Would it make sense to include the two other prevalent types of links, FTP(s) and "mailto:". I think besides HTTP these two would cover the needs of most people and given that we are using a regex if is just a matter of adding these two protocols to the regex the code does not become much harder.